etcdserver/api/v3rpc/watch.go

```diff
@@ -66,6 +70,15 @@ func (ws *watchServer) Watch(stream pb.Watch_WatchServer) error {
 	return sws.recvLoop()
 }
 
+func (sws *serverWatchStream) newResponseHeader() *pb.ResponseHeader {
+	return &pb.ResponseHeader{
+		ClusterId: uint64(sws.server.ClusterID()),
+		MemberId:  uint64(sws.server.ID()),
+		Revision:  sws.watchStream.Revision(),
```

Comment:
This is important since if the rev in the header is the same with the rev in the kv notification, client know that this watcher is in sync. If the rev in the header is greater than the rev in the kv notification, they know that this watch is a unsynced slow one since the kv notification they got is behind current rev. We want this to be accurate if possible.